minoc/src/main/java/org/opencadc/minoc/GetAction.java

```diff
@@ -90,11 +92,18 @@ public GetAction() {
 
     /**
      * Download the artifact or cutouts of the artifact.
-     * @param artifactURI The identifier for the artifact. 
+     * @param artifactURI The identifier for the artifact.
+     * @return The artifact
      */
     @Override
-    public void execute(URI artifactURI) throws Exception {
-        // TODO
+    public Artifact execute(URI artifactURI) throws Exception {
+        Artifact artifact = super.execute(artifactURI);
```

Comment:
This super call to HeadAction to re-use the setHeader calls seems more complex/subtle than just having a static method like HeadAction.writeHeaders(Artifact, SyncOutput) that both doAction methods call. I missed this fact the fuirst read through and was going to comment that you need to set headers because nothing here says "set headers".